test/Deploy.t.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.8.23;
+
+import { Test, console2 } from "../lib/forge-std/src/Test.sol";
+
+import { IMinterGateway } from "../src/interfaces/IMinterGateway.sol";
+import { IMToken } from "../src/interfaces/IMToken.sol";
+import { IEarnerRateModel } from "../src/rateModels/interfaces/IEarnerRateModel.sol";
+import { IMinterRateModel } from "../src/rateModels/interfaces/IMinterRateModel.sol";
+
+import { DeployBase } from "../script/DeployBase.sol";
+
+import { MockTTGRegistrar } from "./utils/Mocks.sol";
+
+contract Deploy is Test, DeployBase {
+    address internal constant _TTG_VAULT = 0xdeaDDeADDEaDdeaDdEAddEADDEAdDeadDEADDEaD;
+
+    MockTTGRegistrar internal _ttgRegistrar;
+
+    function setUp() external {
+        _ttgRegistrar = new MockTTGRegistrar();
+        _ttgRegistrar.setVault(_TTG_VAULT);
+    }
+
+    function test_deploy() external {
+        (address minterGateway_, address minterRateModel_, address earnerRateModel_) = deploy(
+            address(this),
+            2,
+            address(_ttgRegistrar)
+        );
+
+        address mToken_ = getExpectedMToken(address(this), 2);
+
+        // Minter Gateway assertions
+        assertEq(minterGateway_, getExpectedMinterGateway(address(this), 2));
+        assertEq(IMinterGateway(minterGateway_).ttgRegistrar(), address(_ttgRegistrar));
+        assertEq(IMinterGateway(minterGateway_).ttgVault(), _TTG_VAULT);
+        assertEq(IMinterGateway(minterGateway_).mToken(), mToken_);
+
+        // MToken assertions
```

Comment:
We can also assert the `rateModel()`.

Reply:
No, because those only take effect and TTG proposals are executed.